sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -261,12 +262,14 @@ def __internal_api__get_items_as_dataclasses__(
 
                 results.append(reconstructed_item)
 
-                # Break the loop if we have enough samples
+                # Stop the search if we have enough samples
                 if nb_samples is not None and len(results) == nb_samples:
+                    should_search_for_dataset_items = False
```

Comment:
Not related to this line but let's make a copy of dataset_item_ids and call it `dataset_items_ids_left` to not modify the original list and be more explicit with the name

Reply:
Done 🙂